### ISSUE TITLE
"Bad file descriptor" in on_receivable

### ISSUE THREAD

**Issue by ad-m** [boramalper/magnetico#55]
After 4 minutes of continous working magneticod I received following exception:

```
Traceback (most recent call last):
  File "/home/artur/.local/lib/python3.5/site-packages/magneticod/bittorrent.py", line 87, in on_receivable
    received = self.__socket.recv(8192)
OSError: [Errno 9] Bad file descriptor

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./magneticod", line 9, in <module>
    load_entry_point('magneticod==0.3.0', 'console_scripts', 'magneticod')()
  File "/home/artur/.local/lib/python3.5/site-packages/magneticod/__main__.py", line 81, in main
    loop()
  File "/home/artur/.local/lib/python3.5/site-packages/magneticod/__main__.py", line 163, in loop
    key.fileobj.on_receivable()
  File "/home/artur/.local/lib/python3.5/site-packages/magneticod/bittorrent.py", line 97, in on_receivable
    self.when_error()
  File "/home/artur/.local/lib/python3.5/site-packages/magneticod/__main__.py", line 135, in on_peer_error
    selector.unregister(peer)
  File "/usr/lib/python3.5/selectors.py", line 415, in unregister
    key = super().unregister(fileobj)
  File "/usr/lib/python3.5/selectors.py", line 248, in unregister
    key = self._fd_to_key.pop(self._fileobj_lookup(fileobj))
  File "/usr/lib/python3.5/selectors.py", line 224, in _fileobj_lookup
    return _fileobj_to_fd(fileobj)
  File "/usr/lib/python3.5/selectors.py", line 41, in _fileobj_to_fd
    raise ValueError("Invalid file descriptor: {}".format(fd))
ValueError: Invalid file descriptor: -1
```

Here is report of memory usage:

![dane](https://cloud.githubusercontent.com/assets/3618479/25558429/bf3b5a64-2d26-11e7-9458-4b5c43300d4a.png)

I don't see any similiar report. Please inform me if I omit something.

**Comment by ad-m** [comment]
I tested magneticod again.

Here is stacktrace again:
```
Traceback (most recent call last):
  File "/home/artur/.local/lib/python3.5/site-packages/magneticod/bittorrent.py", line 87, in on_receivable
    received = self.__socket.recv(8192)
OSError: [Errno 9] Bad file descriptor

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./magneticod", line 9, in <module>
    load_entry_point('magneticod==0.3.0', 'console_scripts', 'magneticod')()
  File "/home/artur/.local/lib/python3.5/site-packages/magneticod/__main__.py", line 81, in main
    loop()
  File "/home/artur/.local/lib/python3.5/site-packages/magneticod/__main__.py", line 163, in loop
    key.fileobj.on_receivable()
  File "/home/artur/.local/lib/python3.5/site-packages/magneticod/bittorrent.py", line 97, in on_receivable
    self.when_error()
  File "/home/artur/.local/lib/python3.5/site-packages/magneticod/__main__.py", line 135, in on_peer_error
    selector.unregister(peer)
  File "/usr/lib/python3.5/selectors.py", line 415, in unregister
    key = super().unregister(fileobj)
  File "/usr/lib/python3.5/selectors.py", line 248, in unregister
    key = self._fd_to_key.pop(self._fileobj_lookup(fileobj))
  File "/usr/lib/python3.5/selectors.py", line 224, in _fileobj_lookup
    return _fileobj_to_fd(fileobj)
  File "/usr/lib/python3.5/selectors.py", line 41, in _fileobj_to_fd
    raise ValueError("Invalid file descriptor: {}".format(fd))
ValueError: Invalid file descriptor: -1
```

Here is time result:
```
real	11m13.362s
user	5m25.984s
sys	4m13.244s
```

Here is memory usage graph for that session:
![dane](https://cloud.githubusercontent.com/assets/3618479/25558526/0ac4673a-2d29-11e7-8807-0b636f7b01be.png)

**Comment by ad-m** [comment]
Fixed by #56

**Comment by boramalper** [comment]
I guess the issue is resolved with the latest asyncio re-write.
